### PR TITLE
Added Merge Sort [C]

### LINKED ISSUE #17: Merge Sort [C]
https://en.wikipedia.org/wiki/Merge_sort

### PR DESCRIPTION
**Pull request by Monal5031**
Fixed #17 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous timespan of 7 days.

### CODE REVIEW

File: merge_sort/merge_sort.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#include <stdio.h>
+
+void merge(int *arr, int left, int mid, int right) {
+	int i,j,k;
```
Comment (singhpratyush) [inline merge_sort/merge_sort.c]: Add proper spacing.

File: merge_sort/merge_sort.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	int n_left = mid - left + 1;
+	int n_right = right - mid;
+	int left_arr[n_left];
+	int right_arr[n_right];
+	for (i = 0; i < n_left; i++)
+		left_arr[i] = arr[left + i];
+	for (j = 0; j < n_right; j++)
+		right_arr[j] = arr[mid + 1+ j];
```
Comment (singhpratyush) [inline merge_sort/merge_sort.c]: Add proper spacing.

File: merge_sort/merge_sort.c [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		merge_sort(arr, left, mid);
+		merge_sort(arr, mid + 1, right);
+		merge(arr, left, mid, right);
+	}
+}
+
+int main() {
+	int i;
+	int arr[] = {7, 10, 14, 1, 2, 6, 17, 0};
+	int n = sizeof(arr) / sizeof(arr[0]);
```
Comment (mohitkyadav) [inline merge_sort/merge_sort.c]: `int size` sounds good.

File: merge_sort/merge_sort.c [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+		right_arr[j] = arr[mid + 1 + j];
+	i = 0;
+	j = 0;
+	k = left;
+	while (i < n_left && j < n_right) {
+		if (left_arr[i] <= right_arr[j]) {
+			arr[k] = left_arr[i];
+			i++;
+		}
+		else {
```
Comment (mohitkyadav) [inline merge_sort/merge_sort.c]: } else {

Comment (Monal5031) [inline merge_sort/merge_sort.c]: We do that only with `try-catch` block.

**Comment by Monal5031** [comment]
@singhpratyush Please review.

File: merge_sort/merge_sort.c (hunk 4 above)
Comment (singhpratyush) [inline merge_sort/merge_sort.c]: Do we? I feel that else in the same line looks more straight forward while reading the code. You instantly come to know that the if condition is not over yet.

Comment (mohitkyadav) [inline merge_sort/merge_sort.c]: We do...
https://users.ece.cmu.edu/~eno/coding/CCodingStandard.html

screenshot 1

Comment (Monal5031) [inline merge_sort/merge_sort.c]: @singhpratyush It wasn't pointed out in previous algorithms so I thought we aren't following that rule. I'll make the necessary changes.

**Pull request merged**